source/frontend/src/features/properties/map/compensation/models.ts

```diff
@@ -53,9 +69,151 @@ export class CompensationRequisitionFormModel {
     compensation.generationDatetTime = apiModel.generationDate || '';
     compensation.specialInstruction = apiModel.specialInstruction || '';
     compensation.detailedRemarks = apiModel.detailedRemarks || '';
+    compensation.payee = PayeeOption.fromApi(apiModel.payees);
     compensation.isDisabled = booleanToString(apiModel.isDisabled);
     compensation.rowVersion = apiModel.rowVersion ?? null;
 
     return compensation;
   }
 }
+
+enum PayeeType {
+  AcquisitionTeam = 'ACQUISITION_TEAM',
+  OwnerRepresentative = 'OWNER_REPRESENTATIVE',
+  OwnerSolicitor = 'OWNER_SOLICITOR',
+  Owner = 'OWNER',
+}
+
+export class PayeeOption {
+  public readonly api_id: number;
+  public readonly text: string;
+  public readonly value: string;
+  public readonly payeeType: PayeeType;
+
+  private constructor(api_id: number, text: string, value: string, payeeType: PayeeType) {
+    this.api_id = api_id;
+    this.text = text;
+    this.value = value;
+    this.payeeType = payeeType;
+  }
+
+  public static fromApi(payeeModels: Api_CompensationPayee[]): string {
+    if (payeeModels.length === 1) {
+      const apiModel = payeeModels[0];
+      if (apiModel.acquisitionOwnerId) {
+        return PayeeOption.generateKey(apiModel.acquisitionOwnerId, PayeeType.Owner);
+      }
+      if (apiModel.ownerRepresentativeId) {
+        return PayeeOption.generateKey(
+          apiModel.ownerRepresentativeId,
+          PayeeType.OwnerRepresentative,
+        );
+      }
+      if (apiModel.motiSolicitorId) {
+        return PayeeOption.generateKey(apiModel.motiSolicitorId, PayeeType.AcquisitionTeam);
+      }
+      if (apiModel.ownerSolicitorId) {
+        return PayeeOption.generateKey(apiModel.ownerSolicitorId, PayeeType.OwnerSolicitor);
+      }
+    }
+    return '';
+  }
+
+  public static toApi(
+    compensationRequisitionId: number | null,
+    value: string,
+    options: PayeeOption[],
+  ): Api_CompensationPayee | null {
+    if (isNullOrWhitespace(value)) {
+      return null;
+    }
+
+    const payeeOption = options.find(x => x.value === value);
+
+    if (payeeOption === undefined) {
+      return null;
+    }
+
+    const payee: Api_CompensationPayee = {
+      id: 0,
```

Comment:
oh, I understand your service logic now. you aren't passing the compensation payee id up so you need to manually check if the subentity has changed.

Hmm... personally I think I'd rather do some work on the frontend to maintain the rowversion/id rather then the backend. What was your thinking for implementing this way?

Reply:
The problem is that the combobox that holds the payee could be selected and unselected (or the other way around) causing the system to remove the id. I will put extra logic to pass the payee id, but to maintain the pattern we have that I think the backend should not trust the id coming from the frontend anyway

Reply:
I've spent some time on this and I don't think is a good idea to pass the id of the payee if it's going to be validated on the backend again. Let me know if you want to chat about this.